components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/JsGraphBuilder.java

```diff
@@ -1012,6 +1018,24 @@ private JSExecutionMonitorData endScriptExecutionMonitor(String identifier) {
         return getJSExecutionSupervisor().completed(identifier);
     }
 
+    private void setCurrentStepAsSubjectIdentifier(StepConfig stepConfig) {
+        stepNamedMap.forEach((integer, config) -> { // remove existing subject identifier step
+            if (config.isSubjectIdentifierStep()) {
+                config.setSubjectIdentifierStep(false);
+            }
+        });
+        stepConfig.setSubjectIdentifierStep(true);
+    }
+
+    private void setCurrentStepAsSubjectAttribute(StepConfig stepConfig) {
+        stepNamedMap.forEach((integer, config) -> { // remove existing subject attribute step
```

Comment:
`Remove existing subject attribute step.`

Reply:
This has already been fixed with https://github.com/wso2/carbon-identity-framework/pull/4068/files